core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -560,19 +603,23 @@ class ReplicaManager(val config: KafkaConfig,
    * Append messages to leader replicas of the partition, and wait for them to be replicated to other replicas;
    * the callback function will be triggered either when timeout or the required acks are satisfied;
    * if the callback function itself is already synchronized on some object then pass this object to avoid deadlock.
+   * @return Returning a map of successfully appended topic partitions and a flag indicting whether the HWM has been
+   *         incremented. If the caller no longer passed in completeDelayedRequests, the caller is expected to complete
```

Comment:
"If the caller no longer passed in completeDelayedRequests" The caller still passes this in, just as false.